ktxspec.adoc

```diff
@@ -740,14 +740,28 @@ These 3 items combined form a _Data Format Descriptor_
 The full specification for this is found in the _Khronos Data
 Format Specification_ version 1.3 <<KDF13>>.
 
-If the _dfDescriptor_ describes an sRGB transfer function then
-`<<_vkFormat,vkFormat>>` must be one of the _SRGB_ formats.
-
 The _dfDescriptor_ is partially expanded here in order to provide
 sufficient information for a KTX2 file to be parsed without having to
 refer to <<KDF13>>. It consists of one or more _Descriptor Blocks_
 (dfDescriptorBlock).
 
+The following restrictions must be obeyed when setting the fields of the
+DFD and the image data must match the chosen values:
+
+* `color_model` must be `KHR_DF_MODEL_RGBSDA`, `KHR_DF_MODEL_YUVSDA`
+  or one of the block compressed color models, currently
+  `KDF_DF_MODEL_BC1` to `KDF_DF_MODEL_ASTC`.
+* `color_primaries` may be any of the available values since conversion
+  to the display's primaries is relatively straightforward.
+  `KHR_DF_PRIMARIES_BT709/SRGB` is recommended for LDR images.
+  `KHR_DF_PRIMARIES_BT2020` is recommended for HDR images.
```

Comment:
I don't think that LDR and HDR are the correct terms here. Also, listing "recommendation" between the restrictions may be confusing.

Reply:
What terms do you suggest? I don't think having recommendations here is confusing but do you have a suggestion as to where else to put this recommendation?

Reply:
I'd omit this recommendation for now. Do we have a real-world use case for using BT.2020 primaries with GPU textures? 

I'd change "HDR" to "wide color gamut", since "HDR" could mean different things to different developers depending on the context.

Reply:
> Do we have a real-world use case for using BT.2020 primaries with GPU textures?

Yes. As I wrote earlier, Gran Turismo Sport HDR version.

I think wide color gamut and high dynamic range are different potentially overlapping things. So I'm not sure "wide color gamut" is sufficient especially as I'm not sure you can use sRGB/BT.709 primaries for high dynamic range images.

Reply:
After some more digging, I'm sure that this recommendation should be removed as workflows may be very different.
For example, here're some excerpts from NVIDIA's guidelines for working with HDR/WCG/UHD content:

> a HDR image with sRGB chromaticities can appear more richly colored than an SDR image with DCI-P3 primaries

> Given the complexity and cost of developing content for a game, it is generally impractical to expect UHD and sRGB assets, or convert over the art flow to generate most art in these new wider color spaces. 

> Here are the core principles:
>- Author content with sRGB primaries as today
>- Render high quality HDR data using physically-based shading
>- Perform all rendering operations with respect to the sRGB primaries you use today

Reply:
More on this from NVIDIA:
> Rendering using the sRGB primaries is directly related to this. sRGB primaries are what you render with today. If you were to take your sRGB content you create today and convert it into colors described by the BT. 2020 primaries before performing lighting, you’d see much different results. This happens because material interactions are relative to the color space in which they are defined. The modulation operator we all use to compute the interaction between a material and a light source will produce different results if you change the primaries. As a game developer, you don’t want your artists authoring content to see an object looking brown on their monitor, but the same object being green or purple when you run the game on a UHD class display.

So, the ideal workflow would be to use BT.2020 primaries for all color interactions and display the result on a BT.2020-compliant display (AFAIK, the best consumer displays support only DCI-P3 for now). Obviously, actual workflows will be sub-optimal for quite a long time, so such recommendation ("use HDR with BT.2020") doesn't seem very useful.

Reply:
Removed primary recommendations in commit https://github.com/KhronosGroup/KTX-Specification/pull/75/commits/6d5765e3fc392345d73f17f8ed9457d27ccc14b2.